src/popylar_prf/stimulus.py

```diff
@@ -0,0 +1,248 @@
+    dimensions : Sequence[str] or None, optional
+        Names of the grid dimensions (e.g., `["y", "x"]`). Must match the last grid axis.
+
+    Raises
+    ------
+    GridDesignShapeError
+        If the design and grid dimensions do not match.
+    GridDimensionsError
+        If the number of dimensions of the grid except the last does not match the size of the last grid dimension.
+    DimensionLabelsError
+        If the number of dimensions does not match the grid's last dimension.
+
+    Notes
+    -----
+    The shapes of the design and grid must match according to `design.shape[1:] == grid.shape[:-1]`.
+    That is, all design dimensions but the first must have the same size as the grid
+    dimensions excluding the last grid dimension.
+
+    Examples
+    --------
+    Create a stimulus on a 2D grid.
+    >>> import numpy as np
+    >>> from popylar_prf import Stimulus
+    >>> num_frames, width, height = 10, 16, 16
+    >>> design = np.ones((num_frames, width, height))
+    >>> pixel_size = 0.05
+    >>> x = (np.arange(width) - (width - 1) / 2) * pixel_size
+    >>> y = (np.arange(height) - (height - 1) / 2) * pixel_size
+    >>> xv, yv = np.meshgrid(x, y)
+    >>> grid = np.stack((xv, yv), axis=-1)  # shape (height, width, 2)
+    >>> grid = np.stack((xv, yv), axis=-1)  # shape (height, width, 2)
+    >>> # The coordinates of the bottom-left corner:
+    >>> grid[0, 0, :]
+    array([-0.375, -0.375])
+    >>> # The coordinates of the top-right corner:
+    >>> grid[15, 15, :]
+    array([0.375, 0.375])
+    >>> Stimulus(design=design, grid=grid, dimension_labels=["y", "x"])
+    Stimulus(design=array[10, 16, 16], grid=array[16, 16, 2], dimension_labels=['y', 'x'])
+
+    """
+
+    __hash__ = None  # We don't want the object to be hashable because it's mutable
+
+    def __init__(self, design: np.ndarray, grid: np.ndarray, dimension_labels: Sequence[str] | None = None):
+        self.design = design
+        self.grid = grid
+        self.dimension_labels = dimension_labels
+
+        self._check_grid_design_shape()
+        self._check_grid_dimensions()
+        self._check_dimension_labels()
+
+    def _check_grid_design_shape(self) -> None:
+        if not self.design.shape[1:] == self.grid.shape[:-1]:
+            raise GridDesignShapeError(self.design.shape, self.grid.shape)
+
+    def _check_grid_dimensions(self) -> None:
+        if not len(self.grid.shape[:-1]) == self.grid.shape[-1]:
+            raise GridDimensionsError(self.grid.shape)
+
+    def _check_dimension_labels(self) -> None:
+        if self.dimension_labels is not None and not self.grid.shape[-1] == len(self.dimension_labels):
+            raise DimensionLabelsError(len(self.dimension_labels), self.grid.shape[-1])
+
+    def __repr__(self) -> str:
+        return f"""Stimulus(
+            design={np.array_repr(self.design)},
+            grid={np.array_repr(self.grid)},
+            dimension_labels={self.dimension_labels}
+        )"""
```

Comment:
```suggestion
        return f"""{self.__class__.__name__}(
            design={np.array_repr(self.design)},
            grid={np.array_repr(self.grid)},
            dimension_labels={self.dimension_labels}
        )"""
```
this would make the name automatic, and can also be done for the `__str__` method.